agent/app/agent_unix.go

```diff
@@ -0,0 +1,156 @@
+// +build linux
+
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package app
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/aws/amazon-ecs-agent/agent/ec2"
+	"github.com/aws/amazon-ecs-agent/agent/ecscni"
+	"github.com/aws/amazon-ecs-agent/agent/engine"
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
+	"github.com/aws/amazon-ecs-agent/agent/eni/pause"
+	enisetup "github.com/aws/amazon-ecs-agent/agent/eni/setup"
+)
+
+// initPID defines the process identifier for the init process
+const initPID = 1
+
+// awsVPCCNIPlugins is a list of CNI plugins required by the ECS Agent
+// to configure the ENI for a task
+var awsVPCCNIPlugins = []string{ecscni.ECSENIPluginName,
+	ecscni.ECSBridgePluginName,
+	ecscni.ECSIPAMPluginName,
+}
+
+// initializeTaskENIDependencies initializes all of the dependencies required by
+// the Agent to support the 'awsvpc' networking mode. A non nil error is returned
+// if an error is encountered during this process. An additional boolean flag to
+// indicate if this error is considered terminal is also returned
+func (agent *ecsAgent) initializeTaskENIDependencies(state dockerstate.TaskEngineState, taskEngine engine.TaskEngine) (error, bool) {
+	// Check if the Agent process's pid  == 1, which means it's running without an init system
+	if agent.os.Getpid() == initPID {
+		// This is a terminal error. Bad things happen with invoking the
+		// the ENI plugin when there's no init process in the pid namesapce.
+		// Specifically, the DHClient processes that are started as children
+		// of the Agent will not be reaped leading to the ENI device
+		// disappearing until the Agent is killed.
+		return errors.New("agent is not started with an init system"), true
+	}
+
+	// Set VPC and Subnet IDs for the instance
+	if err, ok := agent.setVPCSubnet(); err != nil {
+		return err, ok
```

Comment:
(same as above) Using `ok` here where `true` indicates terminal is pretty confusing to me.